test/runtests.jl

```diff
@@ -830,3 +830,25 @@ end
         @test_throws ArgumentError readdlm(IOBuffer(""), delim = ',', chunksize = 2^63-1)
     end
 end
+
+@testset "colnames" begin
+    @testset "normalize" begin
+        # test unusual or unsafe (for dataframe column names) strings
+        buffer = IOBuffer("""
+        col.1,col|2,col\$3, col4, do, α,_col7
+        1,2,3,4,5,6,7
+        """)
+        df = readcsv(buffer,normalizenames=true)
+        @show df
+        @test names(df) == [:col_1, :col_2, :col_3,:col4,:_do,:α,:_col7]
+    end
+    @testset "don't normalize" begin
+        buffer = IOBuffer("""
+        col1,col2,col3
+        1,2,3
+        """)
+        df = readcsv(buffer,normalizenames=false)
+        @test names(df) == [:col1, :col2, :col3]
```

Comment:
I think these column names are untouched even if `normalizenames = true`. So, this tests nothing. I think you should test the change of the behavior caused by `normalizenames` using the same column names.

Reply:
the test case is enhanced in latest commit